packages/spec/src/v1.ts

```diff
@@ -56,7 +63,7 @@ export interface StandardSuccessOutput<Output> {
   /**
    * The non-existent issues.
    */
-  readonly issues?: null | undefined;
```

Comment:
Since we don't allow any other optional value to be `null`, `issues` on the output should probably not allow it either to avoid mistakes like checking `issues === undefined`, which would work in most other cases.

Reply:
The idea was to have maximum flexibility. Basically, whenever `issues` is a falsy value (except `false`, because Colin didn't like that 🙃), the discriminated union should "know" that the validation was successful. Valibot uses `undefined`, so I am fine with that, but we should consider the flexibility factor.

Reply:
If the goal is just to ensure that `if(issues)` is a safe discriminator, we don't need to include anything other than `undefined` in the type.

The only allowed values for `issues` would be `undefined` or an array, which is always truthy. The fact that a user could still technically provide `false` or `0n` or whatever and it would be treated as having no issues is an implementation detail.

Unless Colin had some other reason for wanting to include `null` specifically?

Reply:
But what if people prefer to explicitly set issues to `null` when there is no issue? Then their type signature is not compatible with Standard Schema, or am I wrong? We were just trying to be more flexible so as not to force schema libraries to change their implementation. Would you prefer to make the spec stricter? I see benefits in both.

Reply:
I think in a case like this, it is better to have a single unambiguous "empty" result.

Since in all of the other similar cases, we allow `undefined` and not `null`, we should do the same here.

You could image the extension to the logic of "what if someone uses X in this case" could be taken very far- `false`, `[]`, etc.

I think if the concern is a conflict with an existing validator API that returns an object with an `issues` key, we should take a similar approach to the other keys in the spec and use `~issues`, but I think adding extra edge cases here just forces our consumers to eventually handle those cases for speculative benefit.

Reply:
There's clearly some tension between trying to expand the type signature as much as possible (so schema library authors can more easily adapt existing methods with minimal additional code) and elegance/usability for consumers (mostly framework authors for projects like tRPC, etc). 

In general, I think its important to keep in mind that the consumers in question are other OSS maintainers, not "regular devs". I don't think average devs should be calling `~validate`. So achieving elegance in the interface is not the #1 priority. 

The most constrictive design constraint is desigining a spec that works with Valibot, since it's designed with bundle size as its primary constraint. I think it's fair to say no other library is as singularly designed around this metric. 

So in my opinion the general principle we should follow is this: we introduce whatever "inelegancies" are necessary to support Valibot, but no more. It's safe to assume other libraries will be fine adding an implementation of `~validate` that differs from their existing method signatures. It's probably +150 bytes at most. We can litigate this heuristic on the call.